tinygrad/helpers.py

```diff
@@ -22,6 +22,10 @@ def make_pair(x:Union[int, Tuple[int, ...]], cnt=2) -> Tuple[int, ...]: return (
 def flatten(l:Iterator): return [item for sublist in l for item in sublist]
 def mnum(i) -> str: return str(i) if i >= 0 else f"m{-i}"
 def fromimport(mod, frm): return getattr(__import__(mod, fromlist=[frm]), frm)
+def merge_dicts(ds):
```

Comment:
Is there a reason you can't use `dict1.update(dict2)` to merge dicts?

Reply:
`update` does not check if value conflicts, added a test case that `(3, vi) @ (vi, 5)` should fail if they hold different values for `vi`

Reply:
Can you add a type signature to this function?